File: testdata/haskell/sample.hs
module Sample (greet, add, Config) where

import Data.List
import qualified Data.Map as Map
import Control.Monad (when, unless)

-- | A configuration type.
data Config = Config
  { name :: String
  , port :: Int
  }

-- | Greet a user.
greet :: String -> String
greet name = "Hello, " ++ name

-- | Add two numbers.
add :: Int -> Int -> Int
add x y = x + y

-- Internal helper.
helper :: String -> IO ()
helper msg = putStrLn msg

File: testdata/ocaml/sample.ml
open Printf
open List

let greet name =
  sprintf "Hello, %s" name

let rec factorial n =
  if n <= 1 then 1
  else n * factorial (n - 1)

let add x y = x + y

type config = {
  name : string;
  port : int;
}

module Utils = struct
  let helper x = x * 2
end
